File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/FormBehaviorType.java
/*
 Copyright Airship and Contributors
 */

package com.urbanairship.android.layout.property;

import java.util.Locale;

import androidx.annotation.NonNull;

import com.urbanairship.json.JsonException;

public enum FormBehaviorType {
    SUBMIT_EVENT("submit_event");

    @NonNull
    private final String value;

    FormBehaviorType(@NonNull String value) {
        this.value = value;
    }

    @NonNull
    public static FormBehaviorType from(@NonNull String value) throws JsonException {
        for (FormBehaviorType type : FormBehaviorType.values()) {
            if (type.value.equals(value.toLowerCase(Locale.ROOT))) {
                return type;
            }
        }
        throw new JsonException("Unknown Form Behavior Type value: " + value);
    }

    @NonNull
    @Override
    public String toString() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/FormBehaviorType.kt
/*
 Copyright Airship and Contributors
 */
package com.urbanairship.android.layout.property

import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

public enum class FormBehaviorType(
    private val value: String
) {
    SUBMIT_EVENT("submit_event");

    override fun toString(): String {
        return name.lowercase()
    }

    public companion object {

        @Throws(JsonException::class)
        public fun from(value: JsonValue): FormBehaviorType {
            val content = value.requireString().lowercase()

            return entries.firstOrNull { it.value == content }
                ?: throw JsonException("Unknown Form Behavior Type value: $value")
        }
    }
}
